sapphire/transformations/celestial.py

```diff
@@ -240,3 +264,91 @@ def galactic_to_equatorial(latitude, longitude, epoch='J2000'):
+
+
+    def equatorial_to_zenithazimuth_astropy(latitude, longitude, utc_timestamp, eq_coordinates):
+        """ Converts iterables of tuples of equatorial to zenithazimuth coordinates
+
+        :param latitude: Latitude in decimal degrees
+        :param longitude: Longitude in decimal degrees
+        :param utc_timestamp: Unix UTC timestamp integer
+        :param eq_coordinates: np.array of tuples (ra, dec) in radians
+        :return: np.array of tuples (zen, az) in radians
+
+        For increased speed using array input is recommended.
+                """
+        hor_coordinates = equatorial_to_horizontal_astropy(latitude, longitude, utc_timestamp, eq_coordinates)
+        # Convert and flip order of coordinates
+        zenaz_coordinates = [(norm_angle(0.5 * np.pi - i[1]), norm_angle(0.5 * np.pi - i[0])) for i in hor_coordinates]
+
+        return np.array(zenaz_coordinates)
+
+
+    def equatorial_to_horizontal_astropy(latitude, longitude, utc_timestamp, eq_coordinates):
+        """ Converts iterables of tuples of equatorial coordinates to horizontal coordinates
+
+        :param latitude: Latitude in decimal degrees
+        :param longitude: Longitude in decimal degrees
+        :param utc_timestamp: Unix UTC timestamp integer
+        :param eq_coordinates: np.array of tuples (ra, dec) in radians
+        :return: np.array of tuples (az, alt) in radians
+
+        For increased speed using array input is recommended.
+        Contrary to the legacy non-astropy function this one does what its name says!
+        """
+        loc = EarthLocation(longitude, latitude)
+        t = Time(datetime.datetime.utcfromtimestamp(utc_timestamp))
+        eq_frame = SkyCoord(eq_coordinates, location=loc, obstime=t, unit=u.rad, frame='icrs')
+        hor_frame = eq_frame.transform_to('altaz')
+
+        return np.array(zip(hor_frame.az.rad, hor_frame.alt.rad))
+
+
+    def horizontal_to_equatorial_astropy(latitude, longitude, utc_timestamp, hor_coordinates):
+        """ Converts iterables of tuples of horizontal coordinates to equatorial coordinates
+
+        :param latitude: Latitude in decimal degrees
+        :param longitude: Longitude in decimal degrees
+        :param utc_timestamp: Unix UTC timestamp integer
+        :param hor_coordinates: np.array of tuples (az, alt) in radians
+        :return: np.array of tuples (ra, dec) in radians
+        """
+        loc = EarthLocation(longitude, latitude)
+        t = Time(datetime.datetime.utcfromtimestamp(utc_timestamp))
+        hor_frame = SkyCoord(hor_coordinates, location=loc, obstime=t, unit=u.rad, frame='altaz')
+        eq_frame = hor_frame.transform_to('icrs')
+
+        return np.array(zip(eq_frame.ra.rad, eq_frame.dec.rad))
+
+
+except ImportError as e:
+    import warnings
+    warnings.warn(str(e)+"\nImport of astropy failed; astropy transformations therefore not available", ImportWarning)
```

Comment:
Missing new line at end of file.

The import of warnings should also be at the top of the module.